### PR TITLE
Showcase - Add `Mock::App` component in showcase to easily mock a product application in a frameless page

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This is a porting of https://github.com/hashicorp/hds-plus/pull/41 in HDS (with some tweaks, that later I'll port back to HDS+).

The idea comes from the observation of how much code @KristinLBradley (and then @philrenaud) had to write (copy&paste) to mock a "product application" in a frameless page.

That's why I've decided to abstract away all that code in a `Mock::App` so that ultimately to render this page:

screenshot_4324

it's enough to write this code:

```
<Mock::App>
  <:main as |M|>
    <M.PageHeader />
    <M.LongTextContent />
  </:main>
</Mock::App>
```
Note: I've used the `Mock` namespace because in the future we may want to mock other complex UI parts (tables with filter and pagination, forms, etc)

### :hammer_and_wrench: Detailed description

In this PR I have:

- added `Mock::App` component for showcase
    - removed an extra file from showcase `types` folder (leftover?)
- converted existing “demo” frameless pages to use `Mock::App`

Preview of the converted pages:
- [AppFrame showcase](https://hds-showcase-git-showcase-add-mock-app-hashicorp.vercel.app/layouts/app-frame)
- [frameless demo-full-app-frame-with-side-nav](https://hds-showcase-git-showcase-add-mock-app-hashicorp.vercel.app/layouts/app-frame/frameless/demo-full-app-frame-with-side-nav)
- [frameless demo-full-app-frame-with-app-header](https://hds-showcase-git-showcase-add-mock-app-hashicorp.vercel.app/layouts/app-frame/frameless/demo-full-app-frame-with-app-header)
- [frameless demo-full-app-frame-with-app-header-and-app-side-nav](https://hds-showcase-git-showcase-add-mock-app-hashicorp.vercel.app/layouts/app-frame/frameless/demo-full-app-frame-with-app-header-and-app-side-nav)

(for the last two, I have used placeholders instead of the `AppHeader/AppSideNav`, for the time being; we can easily add these components to the `Mock::App` and demo pages, if/when we will publish the `AppHeader/AppSideNav` components)

***

### 👀 Component checklist

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: showcase/types/global.d.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: MPL-2.0
  */
+
+import 'ember-source/types';
+export {};
```
[the remark this answers is not in the document]
Comment (didoo) [reply showcase/types/global.d.ts:7]: Same! I don't know, probably coming from some setup copy&pasted (or blueprinted) by @aklkv. He may know better.

File: showcase/app/components/mock/app/header/app-header.gts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        <HdsDropdown @enableCollisionDetection={{true}} as |dd|>
+          <dd.ToggleIcon @icon="help" @text="help menu" />
+          <dd.Title @text="Help & Support" />
+          <dd.Interactive @text="Documentation" @href="#" />
+          <dd.Interactive @text="Tutorials" @href="#" />
+          <dd.Interactive @text="Terraform Provider" @href="#" />
+          <dd.Interactive @text="Changelog" @href="#" />
+          <dd.Separator />
+          <dd.Interactive @text="Create support ticket" @href="#" />
+          <dd.Interactive @text="Give feedback" @href="#" />
```
[the remark this answers is not in the document]
Comment (didoo) [reply showcase/app/components/mock/app/header/app-header.gts:83]: You're absolutely right! Probably an old code copy&pasted. I'll fix it.

[the remark this answers is not in the document]
Comment (alex-ju) [reply showcase/app/components/mock/app/header/app-header.gts]: we have other instances in showcase (I removed them all only from website), so not a blocker at all

[the remark this answers is not in the document]
Comment (didoo) [reply showcase/app/components/mock/app/header/app-header.gts]: Fixed also those ;)

**Comment by KristinLBradley** [comment]
This looks great and should be very convenient to use, thanks!

File: showcase/app/components/mock/app/main/generic-text-content.gts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      nostrum, reiciendis exercitationem quod deserunt inventore, repellendus
+      officia cum temporibus.</HdsTextBody>
+    <HdsTextBody @tag="p" @size="200">Molestias voluptate magni earum unde
+      officia illum doloribus facere natus molestiae quisquam nobis adipisci non
+      distinctio quam asperiores saepe, ab veniam dolor animi sed accusamus
+      nulla nam dolorem.</HdsTextBody>
+    <HdsTextBody @tag="p" @size="200">Quos reprehenderit molestiae veritatis
+      fugiat eligendi temporibus fuga modi, id recusandae dolor mollitia
+      accusantium soluta cumque corporis ipsa tempore amet dolorum.</HdsTextBody>
+  </div>
```
Comment (KristinLBradley) [inline showcase/app/components/mock/app/main/generic-text-content.gts:40]: Initially I viewed the framed examples in a wide browser screen and I was wondering why there seemed to be no margins between the paragraphs. Then I inspected and realized that the text each paragraph contained wasn't long enough to wrap unless I narrowed my browser screen. No big deal as it's just mock content but just something that confused me initially.

Comment (didoo) [reply showcase/app/components/mock/app/main/generic-text-content.gts:43]: I've updated the length of the paragraphs, see if it's better now.

Comment (KristinLBradley) [reply showcase/app/components/mock/app/main/generic-text-content.gts:43]: Thank you!

**Review by KristinLBradley — APPROVED** [review]

**Pull request merged**